### ISSUE TITLE
Audio example: crashes on exit

### ISSUE THREAD

**Issue by LB--** [mosra/magnum-examples#20]
I'm on Windows 10 64-bit compiling with nuwen MinGW GCC 5.3, and my build script looks like [this](https://gist.github.com/LB--/736fb9cf58284f530519).

Every time I close the application, whether by pressing escape or by standard means, it crashes. This is the entire output from the application:
```plain
AL lib: (WW) UpdateDeviceParams: SSE performs best with multiple of 4 update sizes (882)
Audio Renderer: OpenAL Soft by OpenAL Community
OpenAL version: 1.1 ALSOFT 1.17.2
Renderer: AMD Radeon(TM) R7 Graphics by ATI Technologies Inc.
OpenGL version: 4.5.13416 Compatibility Profile Context 15.300.1025.1001
Using optional features:
    GL_ARB_ES2_compatibility
    GL_ARB_direct_state_access
    GL_ARB_get_texture_sub_image
    GL_ARB_invalidate_subdata
    GL_ARB_multi_bind
    GL_ARB_separate_shader_objects
    GL_ARB_texture_storage
    GL_ARB_texture_storage_multisample
    GL_ARB_vertex_array_object
    GL_EXT_direct_state_access
    GL_EXT_texture_filter_anisotropic
    GL_KHR_debug
Using driver workarounds:
    amd-nv-no-forward-compatible-core-context
    no-layout-qualifiers-on-old-glsl
Hrtf status: Audio::Context::HrtfStatus::UnsupportedFormat
Hrtf enabled: false
Hrtf specifier:
```
Running it in GDB just gives `Program received signal SIGSEGV, Segmentation fault.` and a corrupt stack frame.

**Comment by Squareys** [comment]
> Hrtf status: Audio::Context::HrtfStatus::UnsupportedFormat
Hrtf enabled: false
Hrtf specifier:

That's worrying me a bit here. For me on Windows 10 + MinGW-w64 5.1.0 (i686-posix-dwarf-rev0) this works fine :) 
I will see if I can get this to crash by playing around with the hrtf settings a bit.

**Comment by Squareys** [comment]
This does *not* seem to be related to the hrtf settings. Btw: To use hrtf you need to copy contents of `openal-soft/hrtf` to `%appdata%/openal/hrtf` and use headphones.

@mosra I cannot reproduce, could you have a look?

**Comment by mosra** [comment]
yup .. just need to eat something first :D

**Comment by mosra** [comment]
Related to

> Container::Array instances returned from plugin crash on dangling deleter pointer after a plugin is unloaded

in mosra/magnum#108. And the fix is *hard*.

**Comment by LB--** [comment]
> This does _not_ seem to be related to the hrtf settings. Btw: To use hrtf you need to copy contents of `openal-soft/hrtf` to `%appdata%/openal/hrtf` and use headphones.

```plain
Hrtf status: Audio::Context::HrtfStatus::Detected
Hrtf enabled: true
Hrtf specifier: default-44100.mhr
```
Yep, the only difference I notice is that the directional audio is slightly more obvious (or maybe there is no difference and I just can't tell).

**Comment by Squareys** [comment]
@LB-- You should *theoretically* be able to differentiate not only left/right as in stereo but additionally front/back/top/bottom (there is an area in which there is so called "front-back-confusion", where you cannot tell whether a source is coming from the front or back. This also happens IRL, but "experience" is usually able to correct this, example: if you drink a glass of water, the sound will most likely not coming from the back of your head.).
Works best with in-ears.

**Comment by mosra** [comment]
Fixed with mosra/corrade@43397de88223c398b485a512d8c55c64c3b1b4ff. Sorry it took so long.
